### DISCUSSION TITLE
Airfoil is Closed Option (No TE Gap)

### DISCUSSION THREAD

**Discussion by goney3** [jxjo/Xoptfoil2#8] [Ideas]
**Idea request:** Can there be an input option for the output optimization to force a closed airfoil? Currently the optimizations have a TE gap. I couldn't find anything in the documentation explaining how to do this (if the ability already exists). Please and thank you!

**Comment by jxjo** [comment]
Hello Keith, 
... the option to set a TE gap was removed in Xoptfoil2 (as part of  the "keep it simple" effort). 
But you can easily set the Te gap (to 0) with the worker in a little batch job like: 

```
worker    -w set te=0.0  -a mySeed.dat       -o mySeed_te0.dat
xoptfoil2 -i myInput.inp -a mySeed_te0.dat
```

Hope this works for your requirements...

Short remark on TE gap: 
In xfoil there is some kind of switch in the aero calculation if the TE gap is below something like 0.02% chord and above.
In the latter case a panel is added to bridge the gap which will lead to better results ...

I wrote on RCN some [explanations about this artefact](https://www.rc-network.de/threads/projekt-v-jx.11886149/page-5#post-12284246) - section 'Xfoil/Xflr5 Special …' (sorry it's in German - but maybe 'translate' can help)

Best regards 

Jochen

**Reply by goney3** [reply]
The issue is that NASA's OpenVSP https://openvsp.org doesn't like having an open trailing edge (ie. "not water tight") and fails on its calculations unless its closed. I know XFLR5 can still do stability analysis etc without this limitation, so maybe I just need to bug the OpenVSP developers about supporting TE gaps? Thank you for the article, your plane looks FAST! :)

![image](https://github.com/jxjo/Xoptfoil2/assets/19561317/eb170e81-1794-4bbd-97fb-40fd5841e9bf)

**Reply by goney3** [reply]
Actually, my bad... apparently the wings and rudder run fine in the OpenVSP analysis... I guess I shouldn't have included some of the other model elements and one of THOSE was not closed... oops! Sorry Jochen!
